### PR TITLE
Cowswap977/deprecate rinkeby

### PR DESCRIPTION
**Pull request by alfetopito**
# Summary

Removing Rinkeby 

Part of 
1. https://github.com/cowprotocol/cowswap/issues/977
2. https://github.com/cowprotocol/explorer/issues/205

# Testing

Unit tests

Will have to update any consumer when upgrading

### CODE REVIEW

File: src/constants/index.ts [hunk 1]
```diff
@@ -1,14 +1,15 @@
...
 
 const { GPv2Settlement } = JSON.parse(contractNetworks as unknown as string) as typeof contractNetworks
 
-export const GP_SETTLEMENT_CONTRACT_ADDRESS: Partial<Record<number, string>> = {
-  [ChainId.MAINNET]: GPv2Settlement[ChainId.MAINNET].address,
-  [ChainId.RINKEBY]: GPv2Settlement[ChainId.RINKEBY].address,
-  [ChainId.GOERLI]: GPv2Settlement[ChainId.GOERLI].address,
-  [ChainId.GNOSIS_CHAIN]: GPv2Settlement[ChainId.GNOSIS_CHAIN].address,
-}
+export const GP_SETTLEMENT_CONTRACT_ADDRESS = ALL_SUPPORTED_CHAIN_IDS.reduce<Record<number, string>>(
```
Comment (alfetopito) [inline src/constants/index.ts:6]: Made this dynamic so next network added/removed there's one less file to update

File: src/constants/chains.ts [hunk 2]
```diff
@@ -1,13 +1,11 @@
 export enum SupportedChainId {
   MAINNET = 1,
-  RINKEBY = 4,
   GOERLI = 5,
   GNOSIS_CHAIN = 100,
 }
 
 export const ALL_SUPPORTED_CHAIN_IDS: SupportedChainId[] = [
   SupportedChainId.MAINNET,
-  SupportedChainId.RINKEBY,
```
Comment (alfetopito) [inline src/constants/chains.ts:10]: Main changes!

Everything else is derived from this

File: src/api/cow/index.ts [hunk 3]
```diff
@@ -352,10 +357,6 @@ export class CowApi extends BaseApi {
     return finalParams
   }
 
-  protected getApiBaseUrl(): Promise<string> {
-    return super.getApiBaseUrl(this.context.env)
-  }
-
```
Comment (alfetopito) [inline src/api/cow/index.ts:358]: Most of changes to this file is the linter/prettier/ide sorting imports and methods

File: README.md [hunk 4]
```diff
@@ -193,7 +193,7 @@ One easy is to check in the CoW Explorer. You can create a CoW Explorer link if
 
 ```js
 // View order in explorer
-console.log(`https://explorer.cow.fi/rinkeby/orders/${orderId}`)
+console.log(`https://explorer.cow.fi/gc/orders/${orderId}`)
```
Comment (ramirotw) [inline README.md:196]: There are two more places where chainId 4 is used on the README

1. https://github.com/cowprotocol/cow-sdk/pull/76/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5L91
2. https://github.com/cowprotocol/cow-sdk/pull/76/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R301

(you have to expand the file to see it)

**Review by ramirotw — APPROVED** [review]

**Review by anxolin — APPROVED** [review]

**Pull request merged**